signup/index.js

```diff
@@ -0,0 +1,87 @@
+const INPUT_EMAIL = document.querySelector('#signup-email');
+const INPUT_PASSWORD = document.querySelector('#signup-password');
+const INPUT_PASSWORD_CHECK = document.querySelector('#check-password')
+const ERROR_MESSAGE = document.querySelector('.error_message');
+const FORM_ELEMENT = document.querySelector('#signup-form');
+
+INPUT_EMAIL.addEventListener('focusout', handleEmailFocusout);
+INPUT_PASSWORD.addEventListener('focusout', handlePasswordFocusout);
+INPUT_PASSWORD_CHECK.addEventListener('focusout', handlePasswordCheckFocusout);
+FORM_ELEMENT.addEventListener('submit', handleSubmit);
+
+
+// 이메일 유효성 확인
+function emailValidation(input) {
+  const REG_EMAIL = /^([0-9a-zA-Z_\.-]+)@([0-9a-zA-Z_-]+)(\.[0-9a-zA-Z_-]+){1,2}$/;
+  return REG_EMAIL.test(input);
+}
+
+function handleEmailFocusout(e) {
+  if (INPUT_EMAIL.value === '') {
+    INPUT_EMAIL.nextElementSibling.textContent = '이메일을 입력해 주세요.';
+    INPUT_EMAIL.classList.add('input_error');
+
+    return false;
+
+  } else if (INPUT_EMAIL.value !== '') {
+    if (INPUT_EMAIL.value === 'test@codeit.com') {
+      INPUT_EMAIL.nextElementSibling.textContent = '이미 사용 중인 이메일입니다.';
+      INPUT_EMAIL.classList.add('input_error');
+      return false;
+    } else if (!emailValidation(INPUT_EMAIL.value)) {
+      INPUT_EMAIL.nextElementSibling.textContent = '올바른 이메일 주소가 아닙니다.';
+      INPUT_EMAIL.classList.add('input_error');
+      return false;
+    }
+  }
+}
+
+
+// 비밀번호 유효성 확인
+function passwordValidation(input) {
+  const REG_PASSWORD = /^(?=.*[a-zA-Z])(?=.*[0-9]).{8,}$/;
+  return REG_PASSWORD.test(input);
+}
+
+function handlePasswordFocusout(e) {
+  if (!passwordValidation(INPUT_PASSWORD.value)) {
+    INPUT_PASSWORD.nextElementSibling.textContent = '비밀번호는 영문,숫자 조합 8자 이상 입력해주세요.';
+    INPUT_PASSWORD.classList.add('input_error');
+
+    return false;
+  }
+}
```

Comment:
함수의 결과값이 true/false냐에 따라 값이 유효한지 아닌지를 판단하는 함수이죠?
그럼 return true로직도 추가하는게 맞을 것 같아요!

(왜 동작했냐면 submit에서 비교문의 `if (value === false)` 이기 때문입니다. 아무것도 리턴하지 않으면 undefined거든요. `undefined === false`조건은 성립하기 때문에 동작은 할수 있지만 비교문을 `if (!value)`로 변경하면 동작하지 않아요. 무튼 사이드 이펙트가 있을 수 있는 여지가 있기 때문에 명확하게 `return true`를 반환하는게 좋을것 같아요.😊)




(handlePasswordCheckFocusout도 동일)
```suggestion
function handlePasswordFocusout(e) {
  if (!passwordValidation(INPUT_PASSWORD.value)) {
    INPUT_PASSWORD.nextElementSibling.textContent = '비밀번호는 영문,숫자 조합 8자 이상 입력해주세요.';
    INPUT_PASSWORD.classList.add('input_error');

    return false;
  }
  
  return true;
}
```